packages/preferences/src/browser/views/components/preference-array-input.tsx

```diff
@@ -0,0 +1,109 @@
+
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as React from 'react';
+import { Preference } from '../../util/preference-types';
+
+interface ArrayInputProps {
```

Comment:
please study carefully coding guidelines: https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#file-name

It should be called `PreferenceArrayInputProps`